pkg/operator/controllers/checker/ingressreplica_test.go

```diff
@@ -0,0 +1,127 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	operatorfake "github.com/openshift/client-go/operator/clientset/versioned/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+)
+
+func TestIngressReplicaChecker(t *testing.T) {
+	ctx := context.Background()
+
+	var defaultReplica int32 = 2
+	var zeroReplica int32 = 0
+
+	for _, tt := range []struct {
+		name              string
+		aroCluster        *arov1alpha1.Cluster
+		ingressController *operatorv1.IngressController
+		expectedState     operatorv1.OperatorCondition
+		expectedReplica   int32
+	}{
+		{
+			name: "ingress controller has default replicas",
+			aroCluster: &arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+			},
+			ingressController: &operatorv1.IngressController{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default",
+					Namespace: "openshift-ingress-operator",
+				},
+				Spec: operatorv1.IngressControllerSpec{
+					Replicas: &defaultReplica,
+				},
+			},
+			expectedState: operatorv1.OperatorCondition{
+				Type:    arov1alpha1.DefaultIngressReplica,
+				Status:  operatorv1.ConditionTrue,
+				Message: "Default replicas in place",
+				Reason:  "CheckDone",
+			},
+			expectedReplica: 2,
+		},
+		{
+			name: "ingress controller has zero replicas",
```

Comment:
Some additional test case ideas:
-`ingress controller has greater than default replicas`
-`ingress controller replicas could not be read`
-`ingress controller replicas failed to update`
I didn't check code coverage of the current tests, but these were a few that came to mind.